content/webapp/services/prismic/transformers/article-series.ts

```diff
@@ -0,0 +1,70 @@
+import { Query } from '@prismicio/types';
+import { ArticleSeries } from '@weco/common/model/article-series';
+import { isNotUndefined } from '@weco/common/utils/array';
+import { Article } from '../../../types/articles';
+import { Series } from '../../../types/series';
+import { ArticlePrismicDocument } from '../types/articles';
+import { transformArticle } from './articles';
+import { transformQuery } from './paginated-results';
+
+type ArticleSeriesWithArticles = {
+  series: Series;
+  articles: Article[];
+};
+
+export const transformArticleSeries = (
+  seriesId: string,
+  articleQuery: Query<ArticlePrismicDocument>
+): ArticleSeriesWithArticles | undefined => {
+  const articles = transformQuery(articleQuery, transformArticle).results;
+
+  if (articles.length === 0) {
+    return undefined;
+  }
+
+  const series = articles[0].series.find(series => series.id === seriesId);
+
+  // GOTCHA: We should hopefully be good here, as we only ever use this for serials,
+  // which are 6 parts long
+  const titles = articles.map(article => article.title);
+
+  const schedule =
+    series && series.schedule.length > 0
+      ? series.schedule.map(scheduleItem => {
+          const index = titles.indexOf(scheduleItem.title);
+          if (index !== -1 && articles[index]) {
+            return articles[index];
+          }
+
+          return scheduleItem;
+        })
+      : [];
+
+  if (isNotUndefined(series)) {
+    const seriesWithItems: Series = {
+      ...(series as ArticleSeries),
+      // Add some colour
+      items:
+        schedule.length > 0
+          ? schedule.map(item => {
+              return item.type === 'article-schedule-items' ||
```

Comment:
Most of this is copied directly from `article-series.js`; the one change is on this line – TypeScript helpfully told me that this type must be `article-schedule-items` (plural), not `article-schedule-item` (singular), and so the old check would always have returned false.

I can't find anywhere the site is broken because of that bug, but might as well fix it while I'm here.

Reply:
I think probably not for this PR, but this function could do with a refactor - it's quite confusing.

Reply:
I'll stick in a todo note.